.dotnet/src/Generated/Models/MessageCreationOptions.cs

```diff
@@ -20,7 +20,7 @@ internal MessageCreationOptions(MessageRole role, IList<MessageContent> content,
             Metadata = metadata;
             SerializedAdditionalRawData = serializedAdditionalRawData;
         }
-        public IList<MessageCreationAttachment> Attachments { get; }
+        public IList<MessageCreationAttachment> Attachments { get; set; }
```

Comment:
This is another collection that is getting a new setter for some reason (in addition to the Metadata property in other classes). 🤔 Could you add this one to the script?

Reply:
Seems like this not the generator issue. Nullable collections will have generated setters. This is the only case where a collection property has setters. That is why the `Metadata` and `Attachments` properties have setters.

Should we keep the setter for `Metadata` and `Attachments` ?

Reply:
Oh, that is interesting. Then yes, let's keep them.